### ISSUE TITLE
Errors while loading saved state into a new TGN instance

### ISSUE THREAD

**Issue by fwaris** [twitter-research/tgn#3]
I successfully trained a TGN model using my own dataset. However when I try load a saved model into a new TGN instance, I get the following errors:

Exception has occurred: RuntimeError
Error(s) in loading state_dict for TGN:
	Missing key(s) in state_dict: "message_function.mlp.0.weight", "message_function.mlp.0.bias", "message_function.mlp.2.weight", "message_function.mlp.2.bias", "message_function.layers.0.weight", "message_function.layers.0.bias", "message_function.layers.2.weight", "message_function.layers.2.bias". 
	size mismatch for memory_updater.memory_updater.weight_ih: copying a param with shape torch.Size([516, 517]) from checkpoint, the shape in current model is torch.Size([516, 100]).
  File "C:\Users\fwaris\source\repos\tgn-master\tgnexport.py", line 58, in <module>
    tgn.load_state_dict(model_state)

looks like mlp weights are missing and there is a mismatch in shape for 'weight_ih'.

**Comment by fwaris** [comment]
As an alternative, I used a pickled version. It seems to work however the second call to compute_edge_probabilities(...) causes a "Trying to update memory to time in the past" error. 

I don't fully understand the internals of the memory mechanism yet but calling .memory.__init_memory__() after unpickling seems to eliminate the above error. Not sure how it affects the performance of the model, though.

**Comment by emalgorithm** [comment]
It looks like you initially saved a model without memory, and then tried to load its state dict into a model with memory. You can tell this because when you're loading the state dict it is saying that it's missing the memory updater and message function weights, which are present only when memory is enabled. Let me know if this solves the problem.

**Comment by fwaris** [comment]
Thanks for the response. The real issue was that I was not setting message_function="identity" when instantiating an empty TGN. I think it defaults to "mlp" in the model but "identity" in the training script and I had not specified anything at training time.

I would propose to make them consistent across both.

**Comment by Ghadah-Al** [comment]
> I successfully trained a TGN model using my own dataset. However when I try load a saved model into a new TGN instance, I get the following errors:
> 
> Exception has occurred: RuntimeError Error(s) in loading state_dict for TGN: Missing key(s) in state_dict: "message_function.mlp.0.weight", "message_function.mlp.0.bias", "message_function.mlp.2.weight", "message_function.mlp.2.bias", "message_function.layers.0.weight", "message_function.layers.0.bias", 


I tried to improve the accuracy of node classification, but when I used LSTMPOOL instant of TemporalAttentionLayer  I got the same error.
